src/views/forms/AccountManagementFormsNew/WithdrawForm/WithdrawForm.tsx

```diff
@@ -0,0 +1,522 @@
+      setToAddress(e.target.value);
+    },
+    [setToAddress]
+  );
+
+  const onChangeAmount = useCallback(
+    ({ value }: NumberFormatValues) => {
+      setAmount(value);
+      setOnSubmitErrorMessage(undefined);
+    },
+    [setAmount]
+  );
+
+  const onClickMax = useCallback(() => {
+    setAmount(freeCollateralBN.toString());
+  }, [freeCollateralBN, setAmount]);
+
+  useEffect(() => {
+    if (sourceAccount?.walletInfo?.name === WalletType.Privy) {
+      // TODO [onboarding-rewrite]: https://linear.app/dydx/issue/OTE-867/coinbase-withdrawals
+      // abacusStateManager.setTransferValue({
+      //   field: TransferInputField.exchange,
+      //   value: 'coinbase',
+      // });
+    }
+  }, [sourceAccount, nobleChainId, setToChainId]);
+
+  const onSelectNetwork = useCallback(
+    (chainID: string, type: 'chain' | 'exchange') => {
+      if (chainID) {
+        setAmount('');
+        if (type === 'chain') {
+          setToChainId(chainID);
+        }
+      }
+    },
+    [setAmount, setToChainId]
+  );
+
+  const { errorMessage, alertType } = useWithdrawFormValidation({
+    isCctp,
+    debouncedAmountBN,
+    toAddress,
+    isValidDestinationAddress,
+    onSubmitErrorMessage,
+    toChainId,
+    toToken,
+    freeCollateralBN,
+  });
+
+  const isDisabled =
+    !!errorMessage ||
+    !toToken ||
+    (!toChainId && !exchange) ||
+    debouncedAmountBN.isNaN() ||
+    debouncedAmountBN.isZero() ||
+    isSubmitting ||
+    !isValidDestinationAddress;
+
+  const [isPreviewing, setIsPreviewing] = useState(false);
+  if (isPreviewing) {
+    return (
+      <$Form onSubmit={onSubmit}>
+        <div tw="text-color-text-0">
+          Make sure everything looks correct with your withdrawal before confirming
+        </div>
+        <WithdrawButtonAndReceipt
+          withdrawToken={toToken}
+          route={route}
+          isDisabled={isDisabled}
+          isLoading={isSubmitting}
+        />
+
+        <button type="button" onClick={() => setIsPreviewing(false)}>
+          Go Back
+        </button>
+      </$Form>
+    );
+  }
+
+  return (
+    <$Form>
+      <NetworkSelectMenu
+        selectedExchange={exchange ?? undefined}
+        selectedChain={toChainId ?? undefined}
+        onSelect={onSelectNetwork}
+        chains={chainsForNetwork}
+      />
+      <FormInput
+        type={InputType.Text}
+        placeholder={stringGetter({ key: STRING_KEYS.ADDRESS })}
+        onChange={onChangeAddress}
+        value={toAddress ?? ''}
+        label={
+          <span>
+            {stringGetter({ key: STRING_KEYS.DESTINATION })}{' '}
+            {isValidDestinationAddress ? (
+              <Icon
+                iconName={IconName.Check}
+                tw="mx-[1ch] my-0 text-[0.625rem] text-color-success"
+              />
+            ) : null}
+          </span>
+        }
+        validationConfig={
+          toAddress && Boolean(exchange) && !isValidDestinationAddress
```

Comment:
I prefer !!exchange but this works too

Reply:
hmmm okay i'm down. tbh i've always preferred `Boolean` but i think `!!` is more common at this company so it makes sense to match that pattern